octoprint_octorelay/__init__.py

```diff
@@ -349,6 +346,16 @@ def turn_off_pin(self, relay_pin, inverted, cmdOFF):
         self._logger.info("pin: {} turned off".format(relay_pin))
         self.update_ui()
 
+    def turn_on_pin(self, relay_pin, inverted, cmdON):
+        GPIO.setup(relay_pin, GPIO.OUT)
+        # XOR with inverted
+        GPIO.output(relay_pin, inverted == False)
+        GPIO.setwarnings(True)
+        if cmdON:
+            os.system(cmdON)
+        self._logger.info("pin: {} turned on".format(relay_pin))
+        self.update_ui()
```

Comment:
@NDR008 , the function `turn_on_pin` is being called within a cycle introduced in line 309:

https://github.com/borisbu/OctoRelay/blob/cd52345b03b61de12cfd26eb275bf42d0441f47c/octoprint_octorelay/__init__.py#L309

Which makes this statement (updating the UI) being fired for each of the relays in configuration.

Apart from that, the same UI update is fired in line 320 anyway, after finishing the cycle:

https://github.com/borisbu/OctoRelay/blob/cd52345b03b61de12cfd26eb275bf42d0441f47c/octoprint_octorelay/__init__.py#L320

Question: is there any benefit of firing the UI update within the cycle? Can we keep the single UI update in the end of the cycle?

Reply:
```suggestion
```

removing UI update here, since it's called in the end of `print_stopped` which calls this method in a cycle